### PR TITLE
allow custom nodes to load in service mode

### PR DESCRIPTION
**Pull request by mjkkirschner**
in service mode don't access the search model. To me it appears that all the logic in the custom node info updated handler is superfluous in service mode so we just bail. Time will tell :)

### CODE REVIEW

File: src/DynamoCore/Utilities/LuceneSearchUtility.cs [hunk 1]
```diff
@@ -390,7 +390,7 @@ internal string CreateSearchQuery(string[] fields, string SearchTerm, bool IsPac
                     foreach (string s in searchTerm.Split(' '))
                     {
                         //If is a ByEmptySpace search and the split words match with more than MaxNodeNamesRepeated nodes then the word is skipped (otherwise the results will be polluted with hundred of not related nodes)
-                        int nodesFrequency = dynamoModel.SearchModel.Entries.Where(entry => entry.Name.ToLower().Contains(s) && !string.IsNullOrEmpty(s)).Count();
+                        int? nodesFrequency = dynamoModel.SearchModel?.Entries.Where(entry => entry.Name.ToLower().Contains(s) && !string.IsNullOrEmpty(s)).Count();
                         if (nodesFrequency > MaxNodeNamesRepeated) continue;
```
Comment (pinzart90) [inline src/DynamoCore/Utilities/LuceneSearchUtility.cs:394]: wont this throw an exception if nodesFrequency is null ?

Comment (mjkkirschner) [reply src/DynamoCore/Utilities/LuceneSearchUtility.cs:394]: no because nodesFreq is nullable

image

**Review by aparajit-pratap — APPROVED** [review]

**Comment by mjkkirschner** [comment]
I need to investigate if the test failures are real or not - they do involve custom nodes so 👀

**Comment by mjkkirschner** [comment]
@QilongTang @RobertGlobant20 @sm6srw - this had a good build here:
https://master-5.jenkins.autodesk.com/job/Dynamo/job/DynamoSelfServe/job/pullRequestValidation/16578/
but if it turns out it could be contributing to those failures we are seeing then feel free to revert. It appear to me those started last week though, but that doesn't prove this doesn't have an effect as well.

**Pull request merged**